src/main/java/io/fabric8/elasticsearch/plugin/KibanaUserReindexAction.java

```diff
@@ -0,0 +1,103 @@
+/**
+* Copyright (C) 2015 Red Hat, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*         http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package io.fabric8.elasticsearch.plugin;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.support.ActionFilter;
+import org.elasticsearch.action.support.ActionFilterChain;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.get.GetResult;
+
+public class KibanaUserReindexAction implements ActionFilter {
+
+	private final ESLogger logger;
+	
+	@Inject
+	public KibanaUserReindexAction(final Settings settings, final ClusterService clusterService, final Client client) {
+		logger = Loggers.getLogger(KibanaUserReindexAction.class, settings);
+		
+		logger.debug("Initializing KibanaUserReindexAction");
+	}
+
+	@Override
+	public int order() {
+		// We want this to be the last in the chain
+		return Integer.MAX_VALUE;
+	}
+
+	@Override
+	public void apply(String action, ActionRequest request,
+			ActionListener listener, ActionFilterChain chain) {
+		chain.proceed(action, request, listener);
+	}
+
+	@Override
+	public void apply(String action, ActionResponse response,
+			ActionListener listener, ActionFilterChain chain) {
+		
+		logger.debug("Response with Action '{}' and class '{}'", action, response.getClass());
+		
+		if ( response instanceof IndexResponse ) {
+			String index = ((IndexResponse) response).getIndex();
+			if ( isKibanaUserIndex(index) ) {
+				index = ".kibana";
```

Comment:
We should allow this to be configurable since its a configurable setting within a Kibana installation